docs: concepts -- add information about tool calling models, update tools section (#21760)
langchain-ai/langchain: docs/docs/concepts.mdx

Some components LangChain implements, some components we rely on third-party integrations for, and others are a mix.

### Chat models

Language models that use a sequence of messages as inputs and return chat messages as outputs (as opposed to using plain text).
These are traditionally newer models (older models are generally `LLMs`, see above).
Chat models support the assignment of distinct roles to conversation messages, helping to distinguish messages from the AI, users, and instructions such as system messages.

> Although the underlying models are messages in, message out, the LangChain wrappers also allow these models to take a string as input. This makes them interchangeable with LLMs.

Reviewer: nit: not sure if interchangeable implies we can change ChatModels -> LLMs
```suggestion
Although the underlying models are messages in, message out, the LangChain wrappers also allow these models to take a string as input. This means you can easily use chat models in place of LLMs.
```

ChatModels also accept other parameters that are specific to that integration.

> #### Tool Calling

Reviewer: should we consolidate this with the `#functiontool-calling` section further down?

Author: yes good catch, missed that section!

Some chat models have been fine-tuned for tool calling and provide a dedicated API for tool calling. 
Given a list of tools and a prompt, these chat models can request that one or more tools be invoked with given arguments.

Generally, such models are better at tool calling than non-fine-tuned models, and are recommended for use cases that require tool calling.

> LangChain provides a standardized interface for tool calling that makes it work with different models.

Reviewer: ```suggestion
LangChain provides a standardized interface for tool calling that is consistent across different models.
```
